docs/api/expect.md

```diff
@@ -1405,3 +1405,70 @@ Don't forget to include the ambient declaration file in your `tsconfig.json`.
 :::tip
 If you want to know more, checkout [guide on extending matchers](/guide/extending-matchers).
 :::
+
+## expect.addEqualityTesters <Badge type="info">1.2.0+</Badge>
+
+- **Type:** `(tester: Array<Tester>) => void`
+
+You can use this method to define custom matchers to test if two object equals are equal. It is compatible with Jest's `expect.extend`.
+
+```ts
+class AnagramComparator {
+  public word: string
+
+  constructor(word: string) {
+    this.word = word
+  }
+
+  equals(other: AnagramComparator): boolean {
+    const cleanStr1 = this.word.replace(/ /g, '').toLowerCase()
+    const cleanStr2 = other.word.replace(/ /g, '').toLowerCase()
+
+    const sortedStr1 = cleanStr1.split('').sort().join('')
+    const sortedStr2 = cleanStr2.split('').sort().join('')
+
+    return sortedStr1 === sortedStr2
+  }
+}
+
+function createAnagramComparator(word: string) {
+  return new AnagramComparator(word)
+}
+
+function isAnagramComparator(a: unknown): a is AnagramComparator {
+  return a instanceof AnagramComparator
+}
```

Comment:
Btw, I'm not sure the value of having these two utilities. I feel these two indirections might distract users from what's happening in first sight.

How about just inline `instanceof` and `new AnagramComparator` usages below?

Reply:
Removed `createAnagramComparator` as the test is inlined.